core/src/main/scala/org/apache/spark/SecurityManager.scala

```diff
@@ -451,7 +452,12 @@ private[spark] class SecurityManager(
       // For Master/Worker, auth secret is in conf; for Executors, it is in env variable
       Option(sparkConf.getenv(SecurityManager.ENV_AUTH_SECRET))
         .orElse(sparkConf.getOption(SecurityManager.SPARK_AUTH_SECRET_CONF)) match {
-        case Some(value) => value
+        case Some(value) =>
+          if (jFiles.exists(jPaths.get(value))) {
+            HashCodes.fromBytes(jFiles.readAllBytes(jPaths.get(value))).toString
+          } else {
+            value
```

Comment:
If I understand this correctly, if the value is the name of a file, look inside the file for the secret; otherwise, the value itself is the secret?

Reply:
yup